p2p/src/sync/peer.rs

```diff
@@ -160,7 +159,6 @@ where
 
         if self.common_services.has_service(Service::Blocks) {
             self.request_headers().await?;
-            self.last_activity = Some(self.time_getter.get_time());
```

Comment:
I changed when we set `last_activity`. It is now set only _after_ we have resolved a peer action. For example, when a peer sends us a block response, we no longer update the activity immediately, but only after we have processed the response.